docs/proposals/logging_rearch.md

```diff
@@ -0,0 +1,55 @@
+# Re architect origin-aggregated-logging to use updated API objects and Templates to install
+
+## Why?
+
+Currently when we develop features or make changes that require configuration
+updates as well as updates to our images we must coordinate those changes in two
+repositories which can lead to things being out of sync or missed.
+
+We should also take the time to evaluate how we deploy/configure the logging
+stack and incorporate newer API objects where able, such as using Stateful Sets
+to deploy Elasticsearch instead of Deployment Configs.
+
+## How?
+
+Taking from our experiences with supporting operations and customers, we should
+be sure to incorporate those gotchas while minimizing the number of hacky
+approaches we've included over releases.
+
+We will need to create templates that allow granular deployment options (e.g.
+how do we deploy Elasticsearch with stateful sets so that we can have different
```

Comment:
I like the idea of stateful sets assuming the way in which the platform provides the storage is something that is compatible with the way ES functions.  I know @portante has had some issues stating that SS are the wrong solution; they allow for the stranding of data if the cluster size is inappropriately reduced in size.  I believe that is a risk but given the complexity of wiring DC's to storage, I believe it is worth taking on.

Reply:
I wonder if that's something we can attempt to control. I'd like to document those concerns and see if we can come up with a solution to address them